### PR TITLE
Redirect all files in /examples folder to kubernetes/examples repo

### PR DESCRIPTION
**Pull request by sebgoa**
**What this PR does / why we need it**:

Examples are being moved to their own repository: https://github.com/kubernetes/examples

We need to remove them from the main repo , but first we need to keep a redirect.

This is a *big* organizational change, but nothing technical (aside from e2e tests)

**Which issue this PR fixes** 

fixes part of #24343 

**Special notes for your reviewer**:

WIP, I still need to figure out what to do with the BUILD script and tests, plus take care of the e2e tests that use some of these examples.

**release notes**
```release-note
Redirect all examples README to the the kubernetes/examples repo
```

### CODE REVIEW

**Comment by sebgoa** [comment]
cc/ @kubernetes/examples big change coming here

**Comment by ahmetb** [comment]
Sent https://github.com/kubernetes/examples/pull/5, maybe we can merge that one first and then update the paths that are not in `staging/` in a single PR.

**Comment by ahmetb** [comment]
This is gonna be interesting... Have we ever moved out Go code and/or YAML files? Replacing all files with the same test is causing Go builds to break (also would break any tutorials referencing to their YAML files or `kubectl create -f https://...`.

What if we did this: If there's a README.md in the directory, replace it with the migration notice. If there's no README.md in a directory, add a README.md with the migration notice, leave all other files untouched?

**Comment by sebgoa** [comment]
yes, I wanted to get this out to get some feedback and see what all it would break.

File: examples/meteor/README.md
Comment (thockin) [inline examples/meteor/README.md]: Some of these links are broken - I just picked this one at random.

**Comment by sebgoa** [comment]
@ahmetb ok I rebased this and just added a notice in the README.

I think it is a good step to start directing folks to the other repo. Once this gets merged we can clean up further.

**Comment by ahmetb** [comment]
/lgtm

I like how you only modified only the README.md files. Otherwise it would break many docs/tutorials in k8s.io and other places. Perhaps delete the `guidelines.md` as well? Also what do you think of using `git.k8s.io/examples` format instead of github URLs? I think this feature is rather new.

**Comment by ahmetb** [comment]
@sebgoa is this still WIP as the title indicates? If not, we should probably request for some help to move this forward.

From what I can tell there  are already conflicting files (meaning = `examples/` patches are still being merged to this repo instead of https://github.com/kubernetes/examples). We should at least try blocking (non-autogenerated) PRs to `examples/`.

```
examples/cockroachdb/README.md
examples/persistent-volume-provisioning/README.md
```

**Comment by sebgoa** [comment]
PR rebased. We need to get this merged so that people will stop merging examples and move to https://github.com/kubernetes/examples

I will try to join the k8s community hangout to talk about it briefly.

then we will need to:

* deal with the current issues filed for examples
* clean up the build
* remove references to these examples in the API docs
* figure out the docs import of these examples

/assign @brendanburns who is an approver

and let's cc @kubernetes/sig-contributor-experience-bugs since the goal is to break up the repos and gain contribution momentum on the examples.

**Comment by kargakis** [comment]
fyi @kubernetes/sig-testing-misc since this was a discussion topic in the last call

**Comment by grodrigues3** [comment]
@brendanburns PTAL.

**Comment by sebgoa** [comment]
@thockin Hi Tim, sorry to ping you directly, but I am out of ideas on how to get attention to this.

thanks

File: examples/elasticsearch/README.md
Comment (timothysc) [inline examples/elasticsearch/README.md]: I don't understand why there isn't just a single readme at the root, everything else seems redundant.

**Review by timothysc — CHANGES_REQUESTED** [review]

File: examples/elasticsearch/README.md
Comment (sebgoa) [inline examples/elasticsearch/README.md]: @timothysc we went with the same concept than when the docs/devel were moved. We leave breadcrumbs in case people linked directly to the repo. that way they know where to go:

https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md

Comment (timothysc) [inline examples/elasticsearch/README.md]: Well I'm ok with it for now just to get it out of the repo.

**Review by timothysc — APPROVED** [review]

**Comment by timothysc** [comment]
@sebgoa could you resolve the conflicts and I will green light this 2day. 

I want these things out ASAP.

**Comment by sebgoa** [comment]
rebased

File: examples/elasticsearch/README.md
Comment (sebgoa) [inline examples/elasticsearch/README.md]: good I have been trying to get this merged sine May 24.

**Pull request merged**
